File: fizzbuzz/python/emrykrems/FizzBuzz_test.py
from FizzBuzz import FizzBuzz

def test_2_For2():
    assert FizzBuzz(2) == 2

def test_9_For13():
    assert FizzBuzz(13) == 13

def test_Fizz_For3():
    assert FizzBuzz(3) == "Fizz"

def test_Buzz_For5():
    assert FizzBuzz(5) == "Buzz"

def test_Ping_For7():
    assert FizzBuzz(7) == "Ping"

def test_Pong_For11():
    assert FizzBuzz(11) == "Pong"

def test_FizzBuzz_For15():
    assert FizzBuzz(15) == "FizzBuzz"

def test_FizzPing_For21():
    assert FizzBuzz(21) == "FizzPing"

def test_FizzPong_For33():
    assert FizzBuzz(33) == "FizzPong"

def test_BuzzPing_For35():
    assert FizzBuzz(35) == "BuzzPing"

def test_BuzzPong_For55():
    assert FizzBuzz(55) == "BuzzPong"

def test_BuzzPingPong_For385():
    assert FizzBuzz(385) == "BuzzPingPong"

def test_FizzBuzzPing_For105():
    assert FizzBuzz(105) == "FizzBuzzPing"

def test_FizzBuzzPingPong_For1155():
    assert FizzBuzz(1155) == "FizzBuzzPingPong"

File: fizzbuzz/java/mikegotta/FizzBuzzTest.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package fizzbuzz;

import org.junit.Test;
import static org.junit.Assert.*;

/**
 *The FizzBuzzTest class tests all possible combinations of the multiples
 * of 3, 5, 7, and 11 to check that the FizzBuzz program returns the 
 * correct string value for the number sent in.
 */
public class FizzBuzzTest {

    @Test
    public void testMultiplesOf3() {
        FizzBuzz FizzBuzz = new FizzBuzz();
        assertEquals("Fizz", FizzBuzz.fizzBuzz(6));
    }
    
    @Test
    public void testMultiplesOf5() {
        assertEquals("Buzz", FizzBuzz.fizzBuzz(10));
    }
    
    @Test
    public void testMultiplesOf7() {
        assertEquals("Ping", FizzBuzz.fizzBuzz(14));
    }
    
    @Test
    public void testMultiplesOf11() {
        assertEquals("Pong", FizzBuzz.fizzBuzz(22));
    }
    
    @Test
    public void testMultiplesOf3And5() {
        assertEquals("FizzBuzz", FizzBuzz.fizzBuzz(15));
    }
    
    @Test
    public void testMultiplesOf3And7() {
        assertEquals("FizzPing", FizzBuzz.fizzBuzz(21));
    }
    
    @Test
    public void testMultiplesOf3And11() {
        assertEquals("FizzPong", FizzBuzz.fizzBuzz(33));
    }
    
    @Test
    public void testMultiplesOf5And7() {
        assertEquals("BuzzPing", FizzBuzz.fizzBuzz(35));
    }
    
    @Test
    public void testMultiplesOf5And11() {
        assertEquals("BuzzPong", FizzBuzz.fizzBuzz(55));
    }
    
    @Test
    public void testMultiplesOf7And11() {
        assertEquals("PingPong", FizzBuzz.fizzBuzz(77));
    }
    
    @Test
    public void testMultiplesOf3And5And7() {
        assertEquals("FizzBuzzPing", FizzBuzz.fizzBuzz(105));
    }
    
    @Test
    public void testMultiplesOf3And5And11() {
        assertEquals("FizzBuzzPong", FizzBuzz.fizzBuzz(165));
    }
    
    @Test
    public void testMultiplesOf3And7And11() {
        assertEquals("FizzPingPong", FizzBuzz.fizzBuzz(231));
    }
    
    @Test
    public void testMultiplesOf5And7And11() {
        assertEquals("BuzzPingPong", FizzBuzz.fizzBuzz(385));
    }
    
    @Test
    public void testMultiplesOf3And5And7And11() {
        assertEquals("FizzBuzzPingPong", FizzBuzz.fizzBuzz(1155));
    }
    
}

    

